### ISSUE TITLE
bug(Datepicker): mat-datepicker-toggle does not announce expanded/collapsed state for screen readers

### ISSUE THREAD

**Issue by RiinaKuu** [angular/components#30406]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

`mat-datepicker-toggle` button uses `aria-haspopup` but does not inform screen reader users whether the popup is expanded or not.

### Reproduction

Example link: https://material.angular.io/components/datepicker/examples

Steps to reproduce:
- Go to the Angular Material link above (I used Chrome)
- Start the screen reader (I used VoiceOver on MacOS)
- Navigate to the datepicker calendar button
- Realize that it is not informing you whether popup calendar is expanded or collapsed

### Expected Behavior

The button should have `aria-expanded` attribute (which should always be paired if used `aria-haspopup`).

### Actual Behavior

<img width="943" alt="Image" src="https://github.com/user-attachments/assets/dee96738-3726-4d43-b6c5-116257d88f2b" />

### Environment

- Angular: 18.2.13
- CDK/Material: 18.2.14
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): macOS
